caproto/_commands.py

```diff
@@ -99,60 +99,28 @@ def padded_string_payload(payload):
                      )
 
 
-def data_payload(values, data_type, data_count, *, metadata=None):
-    ntype = native_type(data_type)
-    size = ctypes.sizeof(DBR_TYPES[data_type])
-
-    if ntype != data_type:
-        size += (data_count - 1) * ctypes.sizeof(DBR_TYPES[ntype])
-
-    if data_count != 1:
-        # TODO this needs some work
-        assert data_count == len(values)
-        value_payload = b''.join(map(bytes, values))
+def data_payload(data, metadata, data_type, data_count):
+    if isinstance(data, ctypes.BigEndianStructure):
+        # This is already a DBR.
+        encoded_data = data
+    else:
+        # This is a tuple of values to be encoded into a DBR.
+        ntype = native_type(data_type)
+        cls_dict = {'_fields_': [('value', ntype * data_count)]}
+        cls = type(name, (ctypes.BigEndianStructure,), cls_dict)
+        cls(data)  # This is probably wrong in a super dumb way.
+    if isinstance(metamdata, ctypes.BigEndianStructure):
```

Comment:
So `values` here is the array, and `metadata` is the corresponding `DBR_TYPE` to the request, or a namedtuple/etc